enterprise/j2ee.core/src/org/netbeans/api/j2ee/core/Profile.java

```diff
@@ -140,134 +123,23 @@ public String toString() {
-                || "JAKARTA_EE_8_FULL".equals(value)) {
-            return JAKARTA_EE_8_FULL;
-        } else if (JAKARTA_EE_8_WEB.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_8_WEB".equals(value)) {
-            return JAKARTA_EE_8_WEB;
-        } else if (JAKARTA_EE_9_FULL.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_9_FULL".equals(value)) {
-            return JAKARTA_EE_9_FULL;
-        } else if (JAKARTA_EE_9_WEB.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_9_WEB".equals(value)) {
-            return JAKARTA_EE_9_WEB;
-        } else if (JAKARTA_EE_9_1_FULL.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_9_1_FULL".equals(value)) {
-            return JAKARTA_EE_9_1_FULL;
-        } else if (JAKARTA_EE_9_1_WEB.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_9_1_WEB".equals(value)) {
-            return JAKARTA_EE_9_1_WEB;
-        } else if (JAKARTA_EE_10_FULL.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_10_FULL".equals(value)) {
-            return JAKARTA_EE_10_FULL;
-        } else if (JAKARTA_EE_10_WEB.toPropertiesString().equals(valueMinusQuotes)
-                || "JAKARTA_EE_10_WEB".equals(value)) {
-            return JAKARTA_EE_10_WEB;
-        } else {
-          return null;
+        if (value == null) {
+            return null;
         }
-    }
-
-    private static String getProfileVersion(@NonNull Profile profile) {
-        String profileDetails = profile.toPropertiesString();
-        int indexOfDash = profileDetails.indexOf("-");
-        if (indexOfDash != -1) {
-            return profileDetails.substring(0, indexOfDash);
-        }
-        return profileDetails;
-    }
 
-    private static boolean compareWebAndFull(@NonNull Profile profileToCompare, @NonNull Profile comparingVersion) {
-        boolean isThisFullProfile = isFullProfile(profileToCompare);
-        boolean isParamFullProfile = isFullProfile(comparingVersion);
-        if (isThisFullProfile && isParamFullProfile) {
-            // Both profiles are Java EE Full
-            return true;
-        }
-        if (!isThisFullProfile && !isParamFullProfile) {
-            // Both profiles are Java EE Web
-            return true;
-        }
-        if (isThisFullProfile && !isParamFullProfile) {
-            // profileToCompare is Java EE Full profile and comparingVersion is only Java EEWeb profile
-            return true;
-        }
-        return false;
-    }
-
-    private static boolean isFullProfile(@NonNull Profile profile) {
-        final String profileDetails = profile.toPropertiesString();
-        if (profileDetails.indexOf("-") == -1) {
-            return true;
+        String valueMinusQuotes = value;
+        if(value.contains("\"")){
+            valueMinusQuotes = value.replace("\"","");
         }
```

Comment:
nitpick: can be changed to:
`String valueMinusQuotes = value.replace("\"","");`

since `replace` is already a no-op if it doesn't contain the String. Adding the `contains` guard just adds another loop to it.

Reply:
Good idea.